File: server/router/system/sys_organization.go
package system

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type OrganizationRouter struct{}

func (s *OrganizationRouter) InitSysOrganizationRouter(Router *gin.RouterGroup) {
	sysOrganizationRouter := Router.Group("sysOrganization").Use(middleware.OperationRecord())
	sysOrganizationRouterWithoutRecord := Router.Group("sysOrganization")
	sysOrganizationApi := v1.ApiGroupApp.SystemApiGroup.OrganizationApi
	{
		sysOrganizationRouter.POST("createSysOrganization", sysOrganizationApi.CreateSysOrganization)   // 新建单位
		sysOrganizationRouter.DELETE("deleteSysOrganization", sysOrganizationApi.DeleteSysOrganization) // 删除单位
		sysOrganizationRouter.PUT("updateSysOrganization", sysOrganizationApi.UpdateSysOrganization)    // 更新单位
	}
	{
		sysOrganizationRouterWithoutRecord.GET("findSysOrganization", sysOrganizationApi.FindSysOrganization)       // 根据ID获取单位
		sysOrganizationRouterWithoutRecord.GET("getSysOrganizationList", sysOrganizationApi.GetSysOrganizationList) // 分页获取单位列表
		sysOrganizationRouterWithoutRecord.GET("getSysOrganizationTree", sysOrganizationApi.GetSysOrganizationTree) // 获取单位树
	}
}

File: web/src/api/sysOrganization.js
import service from '@/utils/request'

// @Tags SysOrganization
// @Summary 创建单位
// @Router /sysOrganization/createSysOrganization [post]
export const createSysOrganization = (data) => {
  return service({
    url: '/sysOrganization/createSysOrganization',
    method: 'post',
    data
  })
}

// @Tags SysOrganization
// @Summary 删除单位
// @Router /sysOrganization/deleteSysOrganization [delete]
export const deleteSysOrganization = (params) => {
  return service({
    url: '/sysOrganization/deleteSysOrganization',
    method: 'delete',
    params
  })
}

// @Tags SysOrganization
// @Summary 更新单位
// @Router /sysOrganization/updateSysOrganization [put]
export const updateSysOrganization = (data) => {
  return service({
    url: '/sysOrganization/updateSysOrganization',
    method: 'put',
    data
  })
}

// @Tags SysOrganization
// @Summary 用id查询单位
// @Router /sysOrganization/findSysOrganization [get]
export const findSysOrganization = (params) => {
  return service({
    url: '/sysOrganization/findSysOrganization',
    method: 'get',
    params
  })
}

// @Tags SysOrganization
// @Summary 分页获取单位列表
// @Router /sysOrganization/getSysOrganizationList [get]
export const getSysOrganizationList = (params) => {
  return service({
    url: '/sysOrganization/getSysOrganizationList',
    method: 'get',
    params
  })
}

// @Tags SysOrganization
// @Summary 获取单位树
// @Router /sysOrganization/getSysOrganizationTree [get]
export const getSysOrganizationTree = (params) => {
  return service({
    url: '/sysOrganization/getSysOrganizationTree',
    method: 'get',
    params
  })
}
